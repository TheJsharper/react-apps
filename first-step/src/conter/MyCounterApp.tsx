import { useCounter } from "./hooks/useCounter";

export const MyCounterApp = () => {
    const { count, increment, decrement, reset } = useCounter(5);
    return <div style={{ display: 'flex', alignItems: 'center', flexDirection: 'column', marginTop: '20px', marginBottom: '20px' }}>


        <h1>Counter: {count}</h1>
        <div style={{ height: '20px' }}></div>
        <div style={{ display: 'flex', gap: '10px' }}>
            <button onClick={ decrement}>-1</button>
            <button onClick={reset}>Reset</button>
            <button onClick={increment}>+1</button>
        </div>

    </div>;
}