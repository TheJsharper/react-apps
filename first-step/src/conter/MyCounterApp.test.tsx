import { render, screen, fireEvent } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { MyCounterApp } from "./MyCounterApp";

describe("MyCounterApp Testing", () => {
    test("  should render correctly", () => {
        const { container } = render(<MyCounterApp />);
        expect(container).toBeDefined();
    });

    test("  should match snapshot", () => {
        const { container } = render(<MyCounterApp />);
        expect(container).toMatchSnapshot();
    });

    test("  should have buttons", () => {
        render(<MyCounterApp />);

        expect(screen.getByRole("button", { name: "-1" })).toBeDefined();

        expect(screen.getByRole("button", { name: "Reset" })).toBeDefined();

        expect(screen.getByRole("button", { name: "+1" })).toBeDefined();
    });

    test("  should show the initial value of 5", () => {
        render(<MyCounterApp />);
        expect(screen.getByText("Counter: 5")).toBeDefined();
    });

    test("  should increment the counter", async () => {
        render(<MyCounterApp />);
        const button = screen.getByRole("button", { name: "+1" });

        await fireEvent.click(button);

        expect(screen.getByText("Counter: 6")).toBeDefined();
    });

    test("  should decrement the counter", async () => {
        render(<MyCounterApp />);
        const button = screen.getByRole("button", { name: "-1" });

        await fireEvent.click(button);

        expect(screen.getByText("Counter: 4")).toBeDefined();
    });

    test("  should reset the counter", async () => {
        render(<MyCounterApp />);
        const button = screen.getByRole("button", { name: "Reset" });

        await fireEvent.click(button);

        expect(screen.getByText("Counter: 0")).toBeDefined();
    });

    test("  should decrement and reset the counter", async () => {
        render(<MyCounterApp />);
        
        const buttonDecrement = screen.getByRole("button", { name: "-1" });
        
        const buttonReset = screen.getByRole("button", { name: "Reset" });

        await fireEvent.click(buttonDecrement);

        expect(screen.getByText("Counter: 4")).toBeDefined();


        await fireEvent.click(buttonReset);

        expect(screen.getByText("Counter: 0")).toBeDefined();
    });

    test("  should increment and reset the counter", async () => {
        render(<MyCounterApp />);

        const buttonIncrement = screen.getByRole("button", { name: "+1" });
        
        const buttonReset = screen.getByRole("button", { name: "Reset" });


        await fireEvent.click(buttonIncrement);

        expect(screen.getByText("Counter: 6")).toBeDefined();

        await fireEvent.click(buttonReset);

        expect(screen.getByText("Counter: 0")).toBeDefined();
    });

    test("  should increment, decrement and reset the counter", async () => {
        
        render(<MyCounterApp />);
        
        const buttonIncrement = screen.getByRole("button", { name: "+1" });
        
        const buttonDecrement = screen.getByRole("button", { name: "-1" });
        
        const buttonReset = screen.getByRole("button", { name: "Reset" });

        await fireEvent.click(buttonIncrement);
        expect(screen.getByText("Counter: 6")).toBeDefined();

        await fireEvent.click(buttonDecrement);
        expect(screen.getByText("Counter: 5")).toBeDefined();

        await fireEvent.click(buttonReset);
        expect(screen.getByText("Counter: 0")).toBeDefined();
        });

    });