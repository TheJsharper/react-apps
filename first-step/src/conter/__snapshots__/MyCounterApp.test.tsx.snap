// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyCounter >   should match snapshot 1`] = `
<div>
  <div
    style="display: flex; align-items: center; flex-direction: column; margin-top: 20px; margin-bottom: 20px;"
  >
    <h1>
      Counter: 
      5
    </h1>
    <div
      style="height: 20px;"
    />
    <div
      style="display: flex; gap: 10px;"
    >
      <button>
        -1
      </button>
      <button>
        Reset
      </button>
      <button>
        +1
      </button>
    </div>
  </div>
</div>
`;

exports[`MyCounterApp Testing >   should match snapshot 1`] = `
<div>
  <div
    style="display: flex; align-items: center; flex-direction: column; margin-top: 20px; margin-bottom: 20px;"
  >
    <h1>
      Counter: 
      5
    </h1>
    <div
      style="height: 20px;"
    />
    <div
      style="display: flex; gap: 10px;"
    >
      <button>
        -1
      </button>
      <button>
        Reset
      </button>
      <button>
        +1
      </button>
    </div>
  </div>
</div>
`;
